### ISSUE TITLE
terminal function docstring/arguments not shown with shift+tab

### ISSUE THREAD

**Issue by EnviousElitist** [ipython/ipython#10417]
I'm on 5.3.0

```
In [2]: import numpy as np
In [3]: np.array 
```

Using shift+tab does not show any function arguments or docstring in the terminal.  Works fine as a tooltip in jupyter notebook and in qtconsole.  Not sure if this is expected or if there is an alternate method of showing the function arguments/docstring in the terminal, other than the monolithic "np.array?"

Tab completion and argument completion works fine once the open parens is given.

EDIT: just noticed I submitted this to jupyter notebook instead of ipython, but I will leave this open in case anyone wants to comment

**Comment by takluyver** [comment]
That's expected in that shift-tab is a notebook shortcut which we've never tried to add to the terminal interface. We could look at offering extra introspection mechanisms now that we've got the new prompt_toolkit interface in place, though.

**Comment by EnviousElitist** [comment]
Yes, there should be some way to view it as there was in IPython 3, with the now removed tab shortcut.

**Comment by takluyver** [comment]
Tab only ever did tab completion in the terminal.

**Comment by EnviousElitist** [comment]
You're right, it looks like I am misremembering.  Extra introspection mechanisms would be a welcome addition for us terminal users.

**Comment by minrk** [comment]
In the terminal, you would do `np.array?<enter>` to get the same info you get from `shift-tab` in the notebook.
